runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorkerTest.java

```diff
@@ -3081,6 +3146,283 @@ public void testActiveWorkRefresh() throws Exception {
     assertThat(server.numGetDataRequests(), greaterThan(0));
   }
 
+  static class FakeClock implements Supplier<Instant> {
+    private static final FakeClock DEFAULT = new FakeClock();
```

Comment:
It seems this coudl just be instance in the test instead of a static

Reply:
I tried to make the fake clock an instance variable of the outer class. This failed because the tests try to de-/serialize the DoFn, and apparently, FakeClock isn't compatible with that. Any ideas?

Reply:
Hmm I forgot it actually serialized/deserialized, I thought we were just passing the DoFn directly.
I think you could make the clock serializable but if you did the deserialized clock would be a different instance than the one injected by the test. 

Another option would be to have a static method on the SlowDoFn that allows setting a static Clock field that all instances use.  Then each test could set this to it's local FakeClock.  That seems a little nicer than a static default clock since it can be cleared/reset between tests to avoid potential leakage.

Reply:
No more global FakeClock instance. FakeSlowDoFn now sets a private static variable to a clock injected via the constructor. The other two unit tests were even simpler to fix.